src/states/node.rs

```diff
@@ -1756,11 +1859,22 @@ impl Node {
         self.peer_mgr.allow_connect(&their_name)?;
         if let Some(token) = self.peer_mgr.get_connection_token(src, dst, their_public_id) {
             self.crust_service.prepare_connection_info(token);
+            return Ok(());
+        }
+
+        let our_pub_info = if let Some(&PeerState::ConnectionInfoReady(ref our_priv_info)) =
+            self.peer_mgr.get_state_by_name(&their_name) {
+            our_priv_info.to_pub_connection_info()
         } else {
-            trace!("{:?} Already sent connection info to {:?}!",
+            trace!("{:?} Not sending connection info request to {:?}",
+                       self,
+                       their_name);
+            return Ok(());
+        };
+        trace!("{:?} Resending connection info request to {:?}",
                    self,
                    their_name);
-        }
+        let _ = self.do_send_connection_info(our_pub_info, their_public_id, src, dst, None);
```

Comment:
any particular reason we are not returning the result of `do_send_connection_info` directly?